src/elasticarray.jl

```diff
@@ -57,76 +57,132 @@ ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(unde
 Base.convert(::Type{T}, A::AbstractArray) where {T<:ElasticArray} = A isa T ? A : T(A)
 
 
+Base.similar(::Type{ElasticArray{T}}, dims::Dims{N}) where {T,N} = ElasticArray{T}(undef, dims...)
+
+
+@inline function Base.:(==)(A::ElasticArray, B::ElasticArray)
+    return ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+end
 
 
-import Base.==
-(==)(A::ElasticArray, B::ElasticArray) =
-    ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+@inline Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
 
+@inline Base.length(A::ElasticArray) = length(A.data)
 
-Base.parent(A::ElasticArray) = A.data
+@propagate_inbounds Base.getindex(A::ElasticArray, i::Int) = getindex(A.data, i)
 
-Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
-@propagate_inbounds Base.getindex(A::ElasticArray, i::Integer) = getindex(A.data, i)
-@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Integer) = setindex!(A.data, x, i)
-@inline Base.IndexStyle(A::ElasticArray) = IndexStyle(A.data)
+@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Int) = setindex!(A.data, x, i)
 
-Base.length(A::ElasticArray) = length(A.data)
+Base.IndexStyle(::Type{<:ElasticArray}) = IndexLinear()
 
-Base.dataids(A::ElasticArray) = Base.dataids(A.data)
 
 
-@inline function Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N}
-    kernel_size, size_lastdim = _split_resize_dims(A, dims)
+@inline Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N} = resize!(A, dims)
+
+@inline function Base.resize!(A::ElasticArray{T,N}, dims::NTuple{N,Integer}) where {T,N}
+    _, size_lastdim = _split_resize_dims(A, dims)
     resize!(A.data, A.kernel_length.divisor * size_lastdim)
-    A
+    return A
+end
+
+@inline function Base.resize!(A::ElasticArray, size_lastdim::Integer)
```

Comment:
I thought about allowing this a while back, but decided against it - in principle, one could also add an array type that is elastic in other dimensions (see #1), and then having a `resize` that takes only one number for a multi-dim array would be a bit unclean.

However, we could export a `resize_lastdim!` function - doesn't seem to be claimed by any package yet.

Reply:
sounds good!

Reply:
Sorry, did accidentially mark this as resolved.

Reply:
Perhaps we can document it and leave it unexported? That gives us some flexibility in the future.

Reply:
@oschulz should be fixed now

Reply:
Yes, maybe better not not export it for now. If we find out it's frequently used, we can still export it later on (the function name seems to be available).